test/config/test_types.rb

```diff
@@ -5,59 +5,146 @@ class TestConfigTypes < ::Test::Unit::TestCase
   include Fluent
 
   sub_test_case 'Config.size_value' do
-    test 'normal case' do
-      assert_equal(2048, Config.size_value("2k"))
-      assert_equal(2048, Config.size_value("2K"))
-      assert_equal(3145728, Config.size_value("3m"))
-      assert_equal(3145728, Config.size_value("3M"))
-      assert_equal(4294967296, Config.size_value("4g"))
-      assert_equal(4294967296, Config.size_value("4G"))
-      assert_equal(5497558138880, Config.size_value("5t"))
-      assert_equal(5497558138880, Config.size_value("5T"))
-      assert_equal(6, Config.size_value("6"))
-    end
-
-    test 'not assumed case' do
-      assert_equal(6, Config.size_value(6))
-      assert_equal(0, Config.size_value("hoge"))
-      assert_equal(0, Config.size_value(""))
-      assert_equal(0, Config.size_value(nil))
+    data("2k" => [2048, "2k"],
+         "2K" => [2048, "2K"],
+         "3m" => [3145728, "3m"],
+         "3M" => [3145728, "3M"],
+         "4g" => [4294967296, "4g"],
+         "4G" => [4294967296, "4G"],
+         "5t" => [5497558138880, "5t"],
+         "5T" => [5497558138880, "5T"],
+         "6"  => [6, "6"])
+    test 'normal case' do |(expected, val)|
+      assert_equal(expected, Config.size_value(val))
+      assert_equal(expected, Config.size_value(val, { strict: true }))
+    end
+
+    data("integer" => [6, 6],
+         "hoge" => [0, "hoge"],
+         "empty" => [0, ""])
+    test 'not assumed case' do |(expected, val)|
+      assert_equal(expected, Config.size_value(val))
+    end
+
+    test 'nil' do
+      assert_equal(nil, Config.size_value(nil))
+    end
+
+    data("integer" => [6, 6],
+         "hoge" => [Fluent::ConfigError.new("name1: invalid value for Integer(): \"hoge\""), "hoge"],
+         "empty" => [Fluent::ConfigError.new("name1: invalid value for Integer(): \"\""), ""])
+    test 'not assumed case with strict' do |(expected, val)|
+      if expected.kind_of? Exception
+        assert_raise(expected) do
+          Config.size_value(val, { strict: true }, "name1")
+        end
+      else
+        assert_equal(expected, Config.size_value(val, { strict: true }, "name1"))
+      end
+    end
+
+    test 'nil with strict' do
+      assert_equal(nil, Config.size_value(nil, { strict: true }))
+    end
+
+    test 'default' do
+      assert_equal(2, Config.size_value(:default, { default: 2 }))
+    end
+
+    test 'no default' do
+      assert_equal(nil, Config.size_value(:default))
```

Comment:
I removed these tests because getting default value at `Config.*_TYPE` is wrong as I mentioned in https://github.com/fluent/fluentd/pull/2685#discussion_r352413304. Now getting default value is done at section.rb and I added tests for them to test_configurable.rb